bolt-modules/boltlib/lib/puppet/datatypes/resultset.rb

```diff
@@ -3,14 +3,14 @@
 Puppet::DataTypes.create_type('ResultSet') do
   interface <<-PUPPET
     attributes => {
-      'results' => Array[Result],
+      'results' => Variant[Array[Result], Array[ApplyResult]],
```

Comment:
Is there ever a time we might want to mix Results and ApplyResults? Perhaps if you were trying to return a ResultSet that contained the results of a set of operations that were taken on a single node?

Reply:
I agree, we should let ResultSet contain a mix of Result and ApplyResult, i.e. `Array[Variant[Result, ApplyResult]]`.

Reply:
I think that makes it harder to iterate on since they don't have similar methods. My thought was we could change this as needed but I guess which direction is a more breaking change?

Reply:
I think enough methods are shared, and you can use a case statement to differentiate types (I think). But going from this to mixed result types in a resultset is the non-breaking change.

Reply:
harder to iterate over the ResultSet

Reply:
I think it's reasonable to keep it this way and extend it later. That will be a backward-compatible change.